apps/fishing-map/features/datasets/upload/NewTrackDataset.tsx

```diff
@@ -78,35 +81,29 @@ function NewTrackDataset({
     if (file) {
       updateFileData(file)
     } else if (dataset) {
-      console.log('TODO', dataset)
-      setDatasetMetadata((meta) => ({
-        ...meta,
+      setDatasetMetadata({
         id: dataset.id,
         name: dataset.name,
         public: true,
         type: DatasetTypes.UserTracks,
         category: DatasetCategory.Environment,
-        fields: meta?.fields,
+        schema: dataset.schema,
         configuration: {
-          latitude: dataset?.configuration?.latitude,
-          longitude: dataset?.configuration?.longitude,
-          timestamp: dataset?.configuration?.timestamp,
-          geometryType: 'tracks' as DatasetGeometryType,
+          ...dataset.configuration,
         } as DatasetConfiguration,
-      }))
-      // setDatasetMetadata(dataset)
+      })
     }
   }, [dataset, file, updateFileData])
 
   const onConfirmClick = useCallback(() => {
     let file: File | undefined
     if (datasetMetadata) {
+      const config = {
```

Comment:
👏 great idea! this way we can make it retro compatible, I think I'd move this to the `upload/datasets-upload.utils` file